features/Roles.feature

```diff
@@ -0,0 +1,179 @@
+
+  @javascript @common
+  Scenario: Changes can be discarded while creating Role
+    When I start creating new "Role"
+      And I set "Name" to "Test RL"
+      And I click on the edit action bar button "Discard changes"
+    Then I should be on "Roles" page
+      And there's no "Test RL" on "Roles" list
+
+  @javascript @common
+  Scenario: New Role can be created
+    When I start creating new "Role"
+      And I set "Name" to "Test RL"
+      And I click on the edit action bar button "Save"
+    Then I should be on "Role" "Test RL" page
+      And Policies list is empty
+      And Assignments list is empty
+
+  @javascript @common
+  Scenario: I can navigate to Roles through breadcrumb
+    Given I go to "Test RL" "Role" page
+    When I click on "Roles" on breadcrumb
+    Then I should be on "Roles" page
+
+  @javascript @common
+  Scenario: Changes can be discarded while editing Content type
+    Given there's "Test RL" on "Roles" list
+    When I start editing "Role" "Test RL"
+      And I set "Name" to "Test RL edited"
+      And I click on the edit action bar button "Discard changes"
+    Then I should be on "Roles" page
+      And there's "Test RL" on "Roles" list
+      And there's no "Test RL edited" on "Roles" list
+
+  @javascript @common
+  Scenario: Role can be edited
+    Given there's "Test RL" on "Roles" list
+    When I start editing "Role" "Test RL"
+      And I set "Name" to "Test RL edited"
+      And I click on the edit action bar button "Save"
+    Then I should be on "Role" "Test RL edited" page
+      And Policies list is empty
+      And Assignments list is empty
+
+  @javascript @common
+  Scenario: User assignation can be discarded
+    Given there's "Test RL edited" on "Roles" list
+    When I start assigning to "Test RL edited"
+      And I select option "Administrator User" from multiple selection "User"
+      And I select options from multiple selection "Group"
+        | option  |
+        | Editors |
+        | Users   |
+      And I set field "Sections" to "true"
+      And I select option "Media" from multiple selection "Sections"
+      And I select "Media/Images"
+      And I click on the edit action bar button "Discard changes"
+    Then I should be on "Role" "Test RL edited" page
+      And Policies list is empty
+      And Assignments list is empty
+
+  @javascript @common
+  Scenario: User can be assigned to role from the Roles list
+    Given there's "Test RL edited" on "Roles" list
+    When I start assigning to "Test RL edited"
+      And I select options from multiple selection "User"
+      | option             |
+      | Administrator User |
+      | Anonymous User     |
+      And I select option "Editor" from multiple selection "Group"
+      And I set field "Subtree" to "true"
+      And I click "Select Subtree" button
+      And I select "Media/Images"
+      And I click "Select content" button
+      And i click "Confirm" button
+      And I click on the edit action bar button "Save"
```

Comment:
There are seven ands in a row, I would consider going for more declarative steps. But if you insist on going pure imperative style then that's fine by me too.

Reply:
Actually it's just a temporary notation of action that we must take, at the end it'll be aggregated for sure.